bot/kodiak/redis.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import asyncio_redis
+import structlog
+
+from kodiak import app_config as conf
+
+conn: asyncio_redis.Connection | None = None
+
+logger = structlog.get_logger()
+
+
+async def get_conn() -> asyncio_redis.Connection:
+    """
+    FastAPI compatible function for accessing the connection pool
+    see: https://fastapi.tiangolo.com/tutorial/dependencies/#to-async-or-not-to-async
+    """
+    global conn
+    if conn is None:
+        logger.info("creating redis pool...")
+        try:
+            redis_db = int(conf.REDIS_URL.database)
+        except ValueError:
+            redis_db = 0
+        conn = await asyncio_redis.Pool.create(
+            host=conf.REDIS_URL.hostname or "localhost",
+            port=conf.REDIS_URL.port or 6379,
+            password=conf.REDIS_URL.password or None,
+            db=redis_db,
+            poolsize=conf.REDIS_POOL_SIZE,
```

Comment:
I think since this is only going to be used for putting incoming messages on the queue, we may need a different redis pool size env var

Reply:
we definitely do

Reply:
hmm, we use this `get_conn` function in both the http server and the queue consumers, thoughts?

Reply:
I've add a param so they can specify